File: course_android/NeteasyTrain/InterfaceCommunication/app/src/main/java/com/better/learn/interfacecommunication/SecondActivity.java
package com.better.learn.interfacecommunication;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.better.learn.interfacecommunication.function.InterfaceManager;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        findViewById(R.id.txt2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(SecondActivity.this,
                        InterfaceManager.getInstance().invoke("hello", String.class), Toast.LENGTH_SHORT)
                        .show();
            }
        });
    }
}

File: about_android/EventBus/app/src/main/java/better/learn/eventbus/SecondActivity.kt
package better.learn.eventbus

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import better.learn.eventbus.event.MessageEvent
import kotlinx.android.synthetic.main.activity_second.*
import org.greenrobot.eventbus.EventBus

class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        second_event.setOnClickListener {
            EventBus.getDefault().post(MessageEvent())
        }
    }
}
